apps/ledger-live-desktop/src/newArch/features/WalletSync/hooks/useQRCode.ts

```diff
@@ -29,20 +29,22 @@ export function useQRCode() {
   const startQRCodeProcessing = useCallback(() => {
     if (!trustchain || !memberCredentials) return;
 
+    let hasCompleted = false;
+
     setError(null);
     setIsLoading(true);
     createQRCodeHostInstance({
       trustchainApiBaseUrl,
       onDisplayQRCode: url => {
         setUrl(url);
-        setIsLoading(false);
       },
       onDisplayDigits: digits => {
         dispatch(setQrCodePinCode(digits));
         dispatch(setFlow({ flow: Flow.Synchronize, step: Step.PinCode }));
       },
       addMember: async member => {
         await sdk.addMember(trustchain, memberCredentials, member);
+        hasCompleted = true;
```

Comment:
you should prefer to do this after `createQRCodeHostInstance` 's Promise terminates.

NB: it can also fails, so maybe you can `.then( successHandling, errorHandling )` on it (depending if you need to handle the error or not, for instance the websocket can be closed due to network cut )

Reply:
ok, i actually misunderstood the code.

I see that you use the hasCompleted in a later state,
actually you have guarantee that if promise succeed, it means the whole process was successful 🤔

Reply:
You can try opening the QRCode screen on LLD, for example, and do nothing. After a while, you'll see the success screen appear, which means you're in the then